### PR TITLE
Feature 144 visualization fixed x axis

### PR DESCRIPTION
**Pull request by Jdallos**
-Issue when filtering the x axis labels to be filterd by times in the data. This causes console errors and a dashed line to appear on the left hand side of the chart.

### CODE REVIEW

File: src/components/Dashboard.tsx [hunk 1]
```diff
@@ -202,22 +253,25 @@ class Dashboard extends React.Component<Props, State> {
...
             <CartesianGrid strokeDasharray="3 3" />
-            <XAxis 
+            <XAxis
               dataKey="time"
-              tickFormatter={ value => moment(value).format("LT") }
-              interval="preserveStartEnd"
-              padding={{ left: 100, right: 100 }}
+              tickFormatter={value => moment(value).format("LT")}
+              padding={{ right: 100 }}
+              ticks={staticLabels}
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:260]:
```suggestion
              ticks={ staticLabels }
```

File: src/components/Dashboard.tsx [hunk 2]
```diff
@@ -202,22 +253,25 @@ class Dashboard extends React.Component<Props, State> {
...
+              interval={0}
+              angle={320}
+              tickSize={10}
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:263]:
```suggestion
              interval={ 0 }
              angle={ 320 }
              tickSize={ 10 }
```

File: src/components/Dashboard.tsx [hunk 3]
```diff
@@ -202,22 +253,25 @@ class Dashboard extends React.Component<Props, State> {
...
+              labelFormatter={value => moment(value).format("LTS")}
             />
             <Legend />
-            <Line 
+            <Line
               type="monotone"
               dataKey="count"
               stroke="#82ca9d"
-              name={ strings.dashboardCount }
+              name={strings.dashboardCount}
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:274]:
```suggestion
              name={ strings.dashboardCount }
```

File: src/components/Dashboard.tsx [hunk 4]
```diff
@@ -202,22 +253,25 @@ class Dashboard extends React.Component<Props, State> {
...
+              padding={{ right: 100 }}
+              ticks={staticLabels}
+              interval={0}
+              angle={320}
+              tickSize={10}
             />
             <YAxis />
             <Tooltip
-              labelFormatter={ value => moment(value).format("LTS") }
+              labelFormatter={value => moment(value).format("LTS")}
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:267]:
```suggestion
              labelFormatter={ value => moment(value).format("LTS") }
```

File: src/components/Dashboard.tsx (hunk 3 above)
Comment (EVP1989) [reply src/components/Dashboard.tsx:274]: Check similar formatting for rest of the file

File: src/constants/index.tsx [hunk 5]
```diff
@@ -107,4 +107,6 @@ export const EVENT_LIST_FILTERS_UPDATED = "EVENT_LIST_FILTERS_UPDATED";
 export type EVENT_LIST_FILTERS_UPDATED = typeof EVENT_LIST_FILTERS_UPDATED;
 
 export const EVENTS_FOUND = "EVENTS_FOUND";
-export type EVENTS_FOUND = typeof EVENTS_FOUND;
\ No newline at end of file
+export type EVENTS_FOUND = typeof EVENTS_FOUND;
+
+export const X_AXIS_TIMES = [6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21];
```
Comment (EVP1989) [inline src/constants/index.tsx:112]: Add tsdocs

**Review by EVP1989 — COMMENTED** [review]
Added comments

File: src/constants/index.tsx (hunk 5 above)
Comment (Jdallos) [reply src/constants/index.tsx:115]: Done

File: src/components/Dashboard.tsx [hunk 6]
```diff
@@ -141,85 +142,146 @@ class Dashboard extends React.Component<Props, State> {
...
+                top: 5,
+                right: 30,
+                left: 20,
+                bottom: 5
+              }}
+            >
+              <CartesianGrid strokeDasharray="3 3" />
+              <XAxis
+                dataKey="time"
+                tickFormatter={value => moment(value).format("LT")}
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:263]:
```suggestion
                tickFormatter={ value => moment(value).format("LT") }
```

File: src/components/Dashboard.tsx [hunk 7]
```diff
@@ -141,85 +142,146 @@ class Dashboard extends React.Component<Props, State> {
...
+            <LineChart
+              data={ hourlyCumulativeCounts(staticLabels, packingsData) }
+              margin={{
+                top: 5,
+                right: 30,
+                left: 20,
+                bottom: 5
+              }}
+            >
+              <CartesianGrid strokeDasharray="3 3" />
```
Comment (EVP1989) [inline src/components/Dashboard.tsx:260]:
```suggestion
              <CartesianGrid strokeDasharray="3 3"/>
```

**Review by EVP1989 — APPROVED** [review]
Approved with suggestions

**Review by EVP1989 — APPROVED** [review]
Approved

**Pull request merged**